test/test_images/runtime/handlers/file_access_attempt.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"io/ioutil"
+	"os"
+
+	"knative.dev/serving/test/types"
+)
+
+type permissionBits uint32
```

Comment:
https://golang.org/pkg/os/#FileMode should work just as well?

Reply:
Yeah both FileMode and `permissionBits` are both unint32. Reason we went with custom type in this test case was to add the ability to compare permission bits via `hasPermission` function. This makes the code more readable.